src/MSALWrapper/AuthFlow/Web.cs

```diff
@@ -63,62 +54,52 @@ public Web(ILogger logger, Guid clientId, Guid tenantId, IEnumerable<string> sco
+            TokenResult tokenResult = null;
 
             try
             {
                 try
                 {
-                    try
-                    {
-                        var tokenResult = await TaskExecutor.CompleteWithin(
-                            this.logger,
-                            this.silentAuthTimeout,
-                            "Get Token Silent",
-                            (cancellationToken) => this.pcaWrapper.GetTokenSilentAsync(this.scopes, account, cancellationToken),
-                            this.errors)
-                            .ConfigureAwait(false);
-                        tokenResult.SetSilent();
+                    tokenResult = await CachedAuth.GetTokenAsync(
+                        this.logger,
+                        this.scopes,
+                        account,
+                        this.pcaWrapper,
+                        this.errors);
 
+                    if (tokenResult != null)
+                    {
                         return (tokenResult, this.errors);
                     }
-                    catch (MsalUiRequiredException ex)
-                    {
-                        this.errors.Add(ex);
-                        this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
-                        var tokenResult = await TaskExecutor.CompleteWithin(
-                            this.logger,
-                            this.interactiveAuthTimeout,
-                            "Interactive Auth",
-                            (cancellationToken) => this.pcaWrapper
+
+                    Func<System.Threading.CancellationToken, Task<TokenResult>> interactiveAuth = (cancellationToken) =>
+                        this.pcaWrapper
                             .WithPromptHint(this.promptHint)
-                            .GetTokenInteractiveAsync(this.scopes, account, cancellationToken),
-                            this.errors)
-                            .ConfigureAwait(false);
+                            .GetTokenInteractiveAsync(this.scopes, account, cancellationToken);
 
-                        return (tokenResult, this.errors);
-                    }
+                    tokenResult = await TaskExecutor.CompleteWithin(
+                        this.logger,
+                        this.interactiveAuthTimeout,
+                        $"{this.Name()} interactive auth",
+                        interactiveAuth,
+                        this.errors).ConfigureAwait(false);
                 }
                 catch (MsalUiRequiredException ex)
                 {
                     this.errors.Add(ex);
-                    this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
-                    var tokenResult = await TaskExecutor.CompleteWithin(
+                    this.logger.LogDebug($"Initial ${this.Name()} auth failed. Trying again with claims.\n{ex.Message}");

```

Comment:
This log assertion is missing in tests.

Reply:
It always has been, I'd like to forgo adding it here, until after the test revamp since many of the tests will move or go away (Same for the other 3 active PRs to use CachedAuth.GetTokenAsync)

Reply:
Resolving with the follow up item #282